sdio/imx6ull/imx6ull-sdio.c

```diff
@@ -0,0 +1,855 @@
+/*
+ * Phoenix-RTOS
+ *
+ * i.MX 6ULL SDIO driver
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Ziemowit Leszczynski, Artur Miller
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+
+#include <sys/mman.h>
+#include <sys/platform.h>
+#include <sys/threads.h>
+#include <sys/interrupt.h>
+
+#include <phoenix/arch/imx6ull.h>
+
+#include <sdio.h>
+
+
+/* SDIO standard commands */
+#define SDIO_CMD_STATE_IDLE     0  /* set the card into idle state       */
+#define SDIO_CMD_STATE_INACTIVE 15 /* set the card into inactive state   */
+#define SDIO_CMD_SELECT_CARD    7  /* select a card by address           */
+#define SDIO_CMD_RELATIVE_ADDR  3  /* set card's relative address        */
+#define SDIO_CMD_SEND_OP_COND   5  /* inquire about operating conditions */
+#define SDIO_CMD_RW_DIRECT      52 /* direct register I/O operation      */
+#define SDIO_CMD_RW_EXTENDED    53 /* bulk register I/O operation        */
+
+#define DMA_BUFFER_SIZE     2048
+#define THREAD_STACK_SIZE   1024
+#define SDIO_IRQ_STS_CARD   (1 << eventBits[SDIO_EVENT_CARD_IRQ])
+#define SDHC_RETRIES        10
+#define ARRAY_LENGTH(array) (sizeof(array) / sizeof(array[0]))
+
+/* hardware platform specific */
+#define USDHC2_ADDR             0x2194000
+#define USDHC2_IRQ              (32 + 23)
+#define SDHC_SYSCTL_RESET_ALL   (1 << 24)
+#define SDHC_SYSCTL_RESET_CMD   (1 << 25)
+#define SDHC_SYSCTL_RESET_DATA  (1 << 26)
+#define SDHC_SYSCTL_CLK_DIV_8   (0x04 << 8)
+#define SDHC_SYSCTL_CLK_DIV_4   (0x02 << 8)
+#define SDHC_SYSCTL_CLK_DEFAULT 0x8010
+#define SDHC_SYSCTL_RESERVED    0xf
+#define SDHC_INTERRUPT_DEFAULTS 0x107f000f
+#define SDHC_CMD_ERROR          0x107f0000
+
+/* maps interrupt event to bit number in registers */
+static const uint8_t eventBits[19] = { 6, 7, 8, 1, 4, 5, 0, 16, 17, 18, 19, 20, 21, 22, 3, 28, 12, 26, 2 };
+
+/* clang-format off */
+enum { reg_ds_addr = 0, reg_blk_att, reg_cmd_arg, reg_cmd_xfer_typ, reg_cmd_rsp0, reg_cmd_rsp1,
+	reg_cmd_rsp2, reg_cmd_rsp3, reg_data_buff_acc_port, reg_pres_state, reg_prot_ctrl, reg_sys_ctrl,
+	reg_int_status, reg_int_status_en, reg_int_signal_en, reg_autocmd12_err_status, reg_host_ctrl_cap,
+	reg_wtmk_lvl, reg_mix_ctrl, reg_force_event = 20, reg_adma_err_status, reg_adma_sys_addr,
+	reg_dll_ctrl = 24, reg_dll_status, reg_clk_tune_ctrl_status, reg_vend_spec = 48, reg_mmc_boot,
+	reg_vend_spec2, reg_tuning_ctrl };
+/* clang-format on */
+
+static struct {
+	volatile uint32_t *base;
+
+	handle_t cmdLock;
+
+	uint16_t blocksz;
```

Comment:
Those newlines are not necessary.
```suggestion
	volatile uint32_t *base;
	handle_t cmdLock;
	uint16_t blocksz;
```

Reply:
Deleted offending newlines.